ULogViewer/ViewModels/Session.cs

```
@@ -1180,6 +1340,56 @@ void ResetLogProfile()
 		public ICommand ResetLogProfileCommand { get; }
 
 
+		// Save marked logs.
+		void SaveMarkedLogs()
+		{
+			foreach (var fileName in this.markedLogsChangedFilePaths)
+				this.SaveMarkedLogs(fileName);
+			this.markedLogsChangedFilePaths.Clear();
+		}
+
+
+		// Save marked logs.
+		void SaveMarkedLogs(string fileName)
+		{
+			// find marked logs in this file name
+			var markedLogs = this.markedLogs.Where(it => it.FileName == fileName);
+
+			// create marked logs info
+			var markedLogInfos = new List<MarkedLogInfo>().Also(markedLogInfos =>
+			{
+				foreach (var markedLog in markedLogs)
+					markedLog.LineNumber?.Let(lineNumber => markedLogInfos.Add(new MarkedLogInfo(fileName, lineNumber, markedLog.Log.Timestamp)));
+			});
+
+			// save or delete marked file
+			var task = ioTaskFactory.StartNew(() =>
+			{
+				var markedFileName = fileName + markedFileExtension;
```

Need to use ```try``` block to ignore error while saving.